From 3aa6611a0bba2cd3204fa4de8e7a2d7bf57e732a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 6 May 2018 12:18:28 +0100
Subject: [PATCH 2/3] Make main.conf stateless

Also default to AutoEnable to true for automatic controller enabling.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac  | 5 +++++
 src/main.c    | 4 ++++
 src/main.conf | 2 +-
 3 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 84a28b0cf..add4aa581 100644
--- a/configure.ac
+++ b/configure.ac
@@ -493,6 +493,11 @@ AC_DEFINE_UNQUOTED(VENDORDIR, "${path_vendordir}",
 			[Vendor directory for the configuration files])
 AC_SUBST(VENDORDIR, "${path_vendordir}")
 
+vendordir="${datadir}/defaults/bluetooth"
+AC_DEFINE_UNQUOTED(VENDORDIR, "${vendordir}",
+			[Directory for the vendor-supplied configuration files])
+AC_SUBST(VENDORDIR, "${vendordir}")
+
 AC_DEFINE_UNQUOTED(MESH_STORAGEDIR, "${storagedir}/mesh",
 			[Directory for the mesh daemon storage files])
 AC_SUBST(MESH_STORAGEDIR, "${storagedir}/mesh")
diff --git a/src/main.c b/src/main.c
index 323d4ffaa..b5e6e9b9c 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1480,6 +1480,10 @@ int main(int argc, char *argv[])
 
 	main_conf = load_config(option_configfile);
 
+	if (!main_conf) {
+		main_conf = load_config(VENDORDIR "/main.conf");
+	}
+
 	parse_config(main_conf);
 
 	if (connect_dbus() < 0) {
diff --git a/src/main.conf b/src/main.conf
index e9c7552a2..01e9fd6c8 100644
--- a/src/main.conf
+++ b/src/main.conf
@@ -350,7 +350,7 @@
 # AutoEnable defines option to enable all controllers when they are found.
 # This includes adapters present on start as well as adapters that are plugged
 # in later on. Defaults to 'true'.
-#AutoEnable=true
+AutoEnable=true
 
 # Audio devices that were disconnected due to suspend will be reconnected on
 # resume. ResumeDelay determines the delay between when the controller
-- 
2.47.2

